x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/common/top_metrics_agg/components/top_metrics_agg_form.tsx

```diff
@@ -0,0 +1,169 @@
+            onChange({ ...aggConfig, sortField: e.target.value });
+          }}
+          data-test-subj="transformSortFieldTopMetricsLabel"
+        />
+      </EuiFormRow>
+
+      {aggConfig.sortField ? (
+        <>
+          {isSpecialFieldSelected ? null : (
+            <>
+              <EuiFormRow
+                label={
+                  <FormattedMessage
+                    id="xpack.transform.agg.popoverForm.sortDirectionTopMetricsLabel"
+                    defaultMessage="Sort direction"
+                  />
+                }
+              >
+                <EuiButtonGroup
+                  type="single"
+                  legend={i18n.translate(
+                    'xpack.transform.agg.popoverForm.sortDirectionTopMetricsLabel',
+                    {
+                      defaultMessage: 'Sort direction',
+                    }
+                  )}
+                  options={sortDirectionOptions}
+                  idSelected={aggConfig.sortDirection ?? ''}
+                  onChange={(id: string) =>
+                    onChange({ ...aggConfig, sortDirection: id as SortDirection })
+                  }
+                  color="text"
+                />
+              </EuiFormRow>
+
+              <EuiFormRow
+                label={
+                  <FormattedMessage
+                    id="xpack.transform.agg.popoverForm.sortModeTopMetricsLabel"
+                    defaultMessage="Sort mode"
+                  />
+                }
+                helpText={
+                  <FormattedMessage
+                    id="xpack.transform.agg.popoverForm.sortModeTopMetricsHelpText"
+                    defaultMessage="Only relevant if the sorting field is an array"
+                  />
+                }
+              >
+                <EuiButtonGroup
+                  type="single"
+                  legend={i18n.translate(
+                    'xpack.transform.agg.popoverForm.sortModeTopMetricsLabel',
+                    {
+                      defaultMessage: 'Sort mode',
+                    }
+                  )}
+                  options={sortModeOptions}
+                  idSelected={aggConfig.sortMode ?? ''}
+                  onChange={(id: string) => {
+                    onChange({ ...aggConfig, sortMode: id as SortMode });
+                  }}
+                  color="text"
+                />
+              </EuiFormRow>
+
+              {sortFieldType && NUMERIC_TYPES_OPTIONS.hasOwnProperty(sortFieldType) ? (
+                <EuiFormRow
+                  label={
+                    <FormattedMessage
+                      id="xpack.transform.agg.popoverForm.numericSortFieldTopMetricsLabel"
+                      defaultMessage="Numeric field"
+                    />
+                  }
+                >
+                  <EuiSelect
+                    options={NUMERIC_TYPES_OPTIONS[sortFieldType as KbnNumericType].map((v) => ({
+                      text: v,
```

Comment:
Like the options above, should these be internationalized?

Reply:
I believe we should not translate type names